### PR TITLE
chore: update copy and links on the readme

### PR DESCRIPTION
**Pull request by beeme1mr**
## This PR

- improves the copy used on the readme
- updates provider and hook links to use the ecosystem page

### Follow-up Tasks

Update the org-level readme template if the copy changes are approved.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -80,7 +80,7 @@ We publish SBOMs with all of our releases as of 0.3.0. You can find them in Mave
 
 - support for various backend [providers](https://openfeature.dev/docs/reference/concepts/provider)
 - easy integration and extension via [hooks](https://openfeature.dev/docs/reference/concepts/hooks)
-- bool, string, numeric and object flag types
+- bool, string, numeric, and object flag types
```
Comment (toddbaert) [inline README.md:83]: Do we need an oxford comma policy?

Comment (beeme1mr) [reply README.md:83]: I tend to prefer oxford commas, but I can revert this change.

File: README.md [hunk 2]
```diff
@@ -165,11 +165,11 @@ public class MyProvider implements FeatureProvider {
 }
 ```
 
-See [here](https://openfeature.dev/docs/reference/technologies/server/java) for a catalog of available providers.
+See [here](https://openfeature.dev/ecosystem?instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Provider&instant_search%5BrefinementList%5D%5Btechnology%5D%5B0%5D=Java) for a catalog of available providers.
 
 ### Hooks:
 
-Hooks are a mechanism that allow for the addition of arbitrary behavior at well-defined points of the flag evaluation life-cycle. Use cases include validation of the resolved flag value, modifying or adding data to the evaluation context, logging, telemetry, and tracking.
+Hooks are a mechanism that allows for adding arbitrary behavior at well-defined points of the flag evaluation life-cycle. Use cases include validating the resolved flag value, modifying or adding data to the evaluation context, logging, telemetry, and tracking.
```
Comment (toddbaert) [inline README.md:172]:
```suggestion
A hook are a mechanism that allows for adding arbitrary behavior at well-defined points of the flag evaluation life-cycle. Use cases include validating the resolved flag value, modifying or adding data to the evaluation context, logging, telemetry, and tracking.
```

We're mixing pluralization a bit here. We should either do "A hook is a mechanism that allows" or "Hooks are mechanisms that allow".

File: README.md [hunk 3]
```diff
@@ -165,11 +165,11 @@ public class MyProvider implements FeatureProvider {
 }
 ```
 
-See [here](https://openfeature.dev/docs/reference/technologies/server/java) for a catalog of available providers.
+See [here](https://openfeature.dev/ecosystem?instant_search%5BrefinementList%5D%5Btype%5D%5B0%5D=Provider&instant_search%5BrefinementList%5D%5Btechnology%5D%5B0%5D=Java) for a catalog of available providers.
```
Comment (toddbaert) [inline README.md:168]: :sunglasses:

File: README.md [hunk 4]
```diff
@@ -199,11 +199,11 @@ See [here](https://www.javadoc.io/doc/dev.openfeature/sdk/latest/index.html) for
 ## ⭐️ Support the project
 
 - Give this repo a ⭐️!
-- Follow us social media:
+- Follow us on social media:
```
Comment (toddbaert) [inline README.md:202]: Oops. This is wrong in the [template](https://github.com/open-feature/.github/tree/main/templates/READMEs) too.

**Review by toddbaert — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (toddbaert) [reply README.md:83]: I was joking. I don't care at all.

File: README.md (hunk 4 above)
Comment (beeme1mr) [reply README.md:202]: I'll fix the template

Comment (toddbaert) [reply README.md:202]: I did already

**Pull request merged**